lib/view_component/compiler.rb

```diff
@@ -64,10 +64,27 @@ def #{method_name}
       CompileCache.register(component_class)
     end
 
+    def add_template_arguments(template, *args)
+      template = template.to_s
+      raise ArgumentError, "Arguments already defined for template #{template}" if template_arguments.key?(template)
+
+      template_exists = templates.any? { |t| t[:base_name].split(".").first == template }
+      raise ArgumentError, "Template does not exist: #{template}" unless template_exists
```

Comment:
I'm thinking this check should also be the other way around: we should only compile methods for the templates we have described (either via the DSL or the `def render_foo`). WDYT?